### DISCUSSION TITLE
`PRAGMA quick_check` ...

### DISCUSSION THREAD

**Discussion by justinclift** [sqlitebrowser/sqlitebrowser#2783] [Ideas]
Just stumbled over [a new commit](https://github.com/sqlite/sqlite/commit/74e5a5a703ce07ab412ac28ea7a14cf93f59da33) in the SQLite source.

The description sounds interesting:

```
Run a "PRAGMA quick_check", if necessary, on the modified table after
an ALTER TABLE ADD COLUMN to verify that added NOT NULL or CHECK constraints
are satisfied by existing rows.  Abort the ADD COLUMN if not.
```

Wonder if we should add the same PRAGMA check, after adding a column?

**Comment by MKleusberg** [comment]
Looks like they integrated this into the ```ALTER TABLE ADD COLUMN``` code. Because we actually use this to add columns already we should profit from this without any changes in our code :smile:

**Reply by justinclift** [reply]
Awesome. :smile:
